src/meta/server_state.cpp

```diff
@@ -3269,8 +3285,321 @@ void server_state::set_max_replica_count(configuration_set_max_replica_count_rpc
              response.old_max_replica_count,
              new_max_replica_count);
 
-    // TODO: update partition-level and app-level max_replica_count
-    response.err = ERR_OK;
+    do_update_max_replica_count(app, app_id, partition_count, rpc);
+}
+
+// ThreadPool: THREAD_POOL_META_STATE
+void server_state::do_update_max_replica_count(std::shared_ptr<app_state> &app,
+                                               int32_t app_id,
+                                               int32_t partition_count,
+                                               configuration_set_max_replica_count_rpc rpc)
```

Comment:
> `app_id` and `partition_count` are redundant, because it can also get in `app`.

Actually in this function both of `app_id` and `partition_count` are not protected at least by a read lock, so I think we can pass them by parameters ?

Reply:
I think you can add read lock to read them in function `do_update_max_replica_count`, your current implementation also need to add read lock in function `set_max_replica_count` and pass them as parameters.

Reply:
Ok.